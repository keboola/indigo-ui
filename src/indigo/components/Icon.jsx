import React from 'react';
import IconSprite from './../img/symbol/svg/sprite.symbol.svg';

export default React.createClass({
    displayName: 'Icon',
    propTypes: {
        iconClass: React.PropTypes.string,
        className: React.PropTypes.string,
    },
    getAdditionalClasses: function () {
        return typeof this.props.className === "undefined" ? "" : this.props.className
    },
    render: function () {
        return (
            <svg className={"icon " + this.getAdditionalClasses()}>
                <use xlinkHref={IconSprite + "#" + this.props.iconClass}/>
            </svg>
        );
    },
});