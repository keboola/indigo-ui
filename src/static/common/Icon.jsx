import React from 'react';
import IconSprite from '../../../dist/icons/symbol/svg/sprite.symbol.svg';

export default React.createClass({
    displayName: 'Icon',
    propTypes: {
        iconClass: React.PropTypes.string,
        className: React.PropTypes.string,
    },
    render: function () {
        return (
            <svg className={"icon-svg " + this.props.className}>
                <use xlinkHref={IconSprite + "#" + this.props.iconClass}/>
            </svg>
        );
    },
});