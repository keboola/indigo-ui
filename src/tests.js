import renderer from 'react-test-renderer';

export const snapshot = (component) => {
  const tree = renderer
    .create(component)
    .toJSON();
  expect(tree).toMatchSnapshot();
};
