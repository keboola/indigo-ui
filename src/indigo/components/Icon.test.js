import React from 'react';

import { snapshot } from "../../tests";
import Icon from './Icon';

describe('<Icon />', () => {

  it('Icon - application', () => {
    snapshot(<Icon iconClass="application"/>);
  });

  it('Icon - calendar', () => {
    snapshot(<Icon iconClass="calendar"/>);
  });


  it('Icon - check', () => {
    snapshot(<Icon iconClass="check"/>);
  });

});
