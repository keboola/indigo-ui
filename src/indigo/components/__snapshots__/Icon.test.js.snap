// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> Icon - application 1`] = `
<svg
  className="icon"
>
  <use
    xlinkHref="sprite.symbol.svg#application"
  />
</svg>
`;

exports[`<Icon /> Icon - calendar 1`] = `
<svg
  className="icon"
>
  <use
    xlinkHref="sprite.symbol.svg#calendar"
  />
</svg>
`;

exports[`<Icon /> Icon - check 1`] = `
<svg
  className="icon"
>
  <use
    xlinkHref="sprite.symbol.svg#check"
  />
</svg>
`;
